test/test_emulation/__init__.py

```diff
@@ -87,6 +87,23 @@ def udev_rules(self):
             rules_dest.unlink()
             self.reload_udev_rules()
 
+    @pytest.fixture(autouse=True, scope='session')
+    def ratbagd_src(self):
+        ratbagd_src = Path('tools/ratbagd.py')
+        ratbagd_dest = Path('test/test_emulation/ratbagd.py')
+        shutil.copyfile(ratbagd_src, ratbagd_dest)
+
+        yield
+
+        if ratbagd_dest.is_file():
+            ratbagd_dest.unlink()
+
+    @pytest.fixture(autouse=True)
+    @pytest.mark.usesfixtures(['ratbagd_src', 'id'])
+    def ratbagd(self):
+        import test_emulation.ratbagd
+        yield ratbagd.Ratbagd(1)
```

Comment:
I actually had to look up what the 1 refers to. Please change this to

```
yield ratbagd.Ratbagd(api_version=1)
```
Code is only written once, but read a thousand times. So you always optimise for the reader, not the writer.